File: src/Akihabara/External/Glog.cs
using Akihabara.Native;

namespace Akihabara.External
{
    public class Glog
    {
        public enum Severity : int
        {
            Info = 0,
            Warning = 1,
            Error = 2,
            Fatal = 3
        }

        public static void Initialize(string name, string dir)
        {
            UnsafeNativeMethods.google_InitGoogleLogging__PKc(name, dir);
        }

        public static void Log(Severity severity, string str)
        {
            switch (severity)
            {
                case Severity.Info:
                {
                    UnsafeNativeMethods.glog_LOG_INFO__PKc(str).Assert();
                    break;
                }
                case Severity.Warning:
                {
                    UnsafeNativeMethods.glog_LOG_WARNING__PKc(str).Assert();
                    break;
                }
                case Severity.Error:
                {
                    UnsafeNativeMethods.glog_LOG_ERROR__PKc(str).Assert();
                    break;
                }
                case Severity.Fatal:
                {
                    UnsafeNativeMethods.glog_LOG_FATAL__PKc(str).Assert();
                    break;
                }
                default:
                {
                    // just to appease Roslyn and ReSharper
                    UnsafeNativeMethods.glog_LOG_INFO__PKc(str);
                    break;
                }
            }
        }
    }
}

File: mediapipe_api/external/glog.cc
// Copyright (c) 2021 homuler
//
// Use of this source code is governed by an MIT-style
// license that can be found in the LICENSE file or at
// https://opensource.org/licenses/MIT.

#include "mediapipe_api/external/glog.h"

void glog_LOG_INFO__PKc(const char* str) {
  LOG(INFO) << str;
}

void glog_LOG_WARNING__PKc(const char* str) {
  LOG(WARNING) << str;
}

void glog_LOG_ERROR__PKc(const char* str) {
  LOG(ERROR) << str;
}

void glog_LOG_FATAL__PKc(const char* str) {
  LOG(FATAL) << str;
}

void google_FlushLogFiles(google::LogSeverity severity) {
  google::FlushLogFiles(severity);
}
